pkg/apiserver/handlers/featuregates/handler_test.go

```diff
@@ -51,30 +50,33 @@ func Test_getGatesResponse(t *testing.T) {
 				},
 			},
 			want: []Response{
+				{Component: "agent", Name: "AntreaIPAM", Status: "Disabled", Version: "ALPHA"},
 				{Component: "agent", Name: "AntreaPolicy", Status: "Disabled", Version: "BETA"},
 				{Component: "agent", Name: "AntreaProxy", Status: "Enabled", Version: "BETA"},
+				{Component: "agent", Name: "CleanupStaleUDPSvcConntrack", Status: "Disabled", Version: "ALPHA"},
 				{Component: "agent", Name: "Egress", Status: egressStatus, Version: "BETA"},
 				{Component: "agent", Name: "EndpointSlice", Status: "Enabled", Version: "BETA"},
-				{Component: "agent", Name: "AntreaIPAM", Status: "Disabled", Version: "ALPHA"},
-				{Component: "agent", Name: "Traceflow", Status: "Enabled", Version: "BETA"},
+				{Component: "agent", Name: "ExternalNode", Status: "Disabled", Version: "ALPHA"},
 				{Component: "agent", Name: "FlowExporter", Status: "Disabled", Version: "ALPHA"},
+				{Component: "agent", Name: "IPsecCertAuth", Status: "Disabled", Version: "ALPHA"},
+				{Component: "agent", Name: "L7NetworkPolicy", Status: "Disabled", Version: "ALPHA"},
+				{Component: "agent", Name: "LoadBalancerModeDSR", Status: "Disabled", Version: "ALPHA"},
+				{Component: "agent", Name: "Multicast", Status: multicastStatus, Version: "BETA"},
+				{Component: "agent", Name: "Multicluster", Status: "Disabled", Version: "ALPHA"},
 				{Component: "agent", Name: "NetworkPolicyStats", Status: "Enabled", Version: "BETA"},
 				{Component: "agent", Name: "NodePortLocal", Status: "Enabled", Version: "BETA"},
-				{Component: "agent", Name: "Multicast", Status: multicastStatus, Version: "BETA"},
+				{Component: "agent", Name: "SecondaryNetwork", Status: "Disabled", Version: "ALPHA"},
 				{Component: "agent", Name: "ServiceExternalIP", Status: "Disabled", Version: "ALPHA"},
-				{Component: "agent", Name: "Multicluster", Status: "Disabled", Version: "ALPHA"},
+				{Component: "agent", Name: "SupportBundleCollection", Status: "Disabled", Version: "ALPHA"},
+				{Component: "agent", Name: "TopologyAwareHints", Status: "Enabled", Version: "BETA"},
+				{Component: "agent", Name: "Traceflow", Status: "Enabled", Version: "BETA"},
+				{Component: "agent", Name: "TrafficControl", Status: "Disabled", Version: "ALPHA"},
 			},
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			got := getAgentGatesResponse(tt.cfg)
-			sort.SliceStable(got, func(i, j int) bool {
-				return got[i].Name < got[j].Name
-			})
-			sort.SliceStable(tt.want, func(i, j int) bool {
-				return tt.want[i].Name < tt.want[j].Name
-			})
+			got := getFeatureGatesResponse(tt.cfg, agentMode)
 			if !reflect.DeepEqual(got, tt.want) {
 				t.Errorf("getAgentGatesResponse() = %v, want %v", got, tt.want)
 			}
```

Comment:
Use `assert.EqualValues(t, got, tt.want)` since it can be easier to find the differences.

Reply:
fix: https://github.com/antrea-io/antrea/pull/5314/commits/0c2ea58435b8b31f6bf14f84ea90727fc7bbaf20